### PR TITLE
Use composer/xdebug-handler v3

### LINKED ISSUE #6213: Allow composer/xdebug-handler v3
https://github.com/composer/xdebug-handler/releases/tag/3.0.0

### PR DESCRIPTION
**Pull request by gharlan**
closes #6213

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -18,7 +18,7 @@
         "ext-json": "*",
         "ext-tokenizer": "*",
         "composer/semver": "^3.2",
-        "composer/xdebug-handler": "^2.0",
+        "composer/xdebug-handler": "^2.0 || ^3.0",
```
Comment (keradus) [inline composer.json:21]: can we drop v2 pls?

Comment (gharlan) [inline composer.json:21]:
```suggestion
        "composer/xdebug-handler": "^3.0",
```

**Review by SpacePossum — APPROVED** [review]

**Comment by stof** [comment]
To reduce the risk of dependency conflicts, I suggest supporting both 2 and 3, as there does not seem to be any change needed in php-cs-fixer between those versions.

**Comment by keradus** [comment]
I'm happy to merge v3 only and reconsider support for v2 if we would see complaints and ppl not updating because of that. ultimately, upcoming shim package would solve the dependencies conflicts once and for all

**Comment by SpacePossum** [comment]
Thank you @gharlan.

**Comment by ankorstore-haddowg** [comment]
@keradus @SpacePossum please add back support for v2 here.
i am unable to upgrade as the composer/composer package itself has not yet updated its dependency on xdebug-handler and thats a dependency of phpstan which we also use.
looks like there is no reason for this not to support 2 also as the only reason 3 was a major is that it removed compatibility for older php version which this package already no longer supports.

**Comment by SpacePossum** [comment]
hi, can you maybe make a PR of it?

**Comment by keradus** [comment]
as alternative - would a shim package fix the situation here?

**Pull request closed without merging**